### PR TITLE
LF: drop support for Exercise's actors

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This is a first set toward dropping support of deprecated LF version
(#7155).

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/CheckAuthorization.scala [hunk 1]
```diff
@@ -86,20 +86,10 @@ private[lf] object CheckAuthorization {
...
       auth: Authorize,
   ): List[FailedAuthorization] = {
-    val controllersDifferFromActors = ex.controllers != ex.actingParties
     authorize(
       passIf = ex.actingParties.nonEmpty,
       failWith = FailedAuthorization.NoControllers(ex.templateId, ex.choiceId, ex.optLocation),
     ) ++
-      authorize(
-        passIf = !controllersDifferFromActors,
-        failWith = FailedAuthorization.ActorMismatch(
```
Comment (hurryabit) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/CheckAuthorization.scala:96]: We can drop the `ActorMismatch` class now as well, can't we?

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala [hunk 2]
```diff
@@ -1097,6 +1097,9 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
             arg = decodeExpr(create.getExpr, definition))
 
         case PLF.Update.SumCase.EXERCISE =>
+          if (versionIsOlderThan(LV.Features.noExerciseActor)) {
```
Comment (hurryabit) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala:1100]: Wouldn't it make sense to check for the minimal supported version before we even start decoding a package?

**Review by hurryabit — APPROVED** [review]
🔥

Nice. Thank you!

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala:1100]: Good point.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/CheckAuthorization.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/CheckAuthorization.scala:96]: good point.

**Pull request merged**